### PR TITLE
registry/filesystem: mention umask

### PR DESCRIPTION
**Pull request by AkihiroSuda**
https://github.com/docker/distribution/pull/2525

### Proposed changes

registry/filesystem: mention umask for security reason

### Related issues (optional)

Discussed in https://github.com/docker/distribution/pull/2525

@dmp42 @stevvooe

### CODE REVIEW

**Comment by dmp42** [comment]
Thanks @AkihiroSuda

File: registry/storage-drivers/filesystem.md [hunk 1]
```diff
@@ -10,7 +10,9 @@ An implementation of the `storagedriver.StorageDriver` interface which uses the
 
 * `rootdirectory`: (optional) The absolute path to a root directory tree in which
 to store all registry files. The registry stores all its data here so make sure
-there is adequate space available. Defaults to `/var/lib/registry`.
+there is adequate space available. Defaults to `/var/lib/registry`. If the directory
+does not exist, it will be created with the `0777` permission. Make sure to use `umask`
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply registry/storage-drivers/filesystem.md]: updated

File: registry/storage-drivers/filesystem.md [hunk 2]
```diff
@@ -10,7 +10,8 @@ An implementation of the `storagedriver.StorageDriver` interface which uses the
 
 * `rootdirectory`: (optional) The absolute path to a root directory tree in which
 to store all registry files. The registry stores all its data here so make sure
-there is adequate space available. Defaults to `/var/lib/registry`.
+there is adequate space available. Defaults to `/var/lib/registry`. If the directory
+does not exist, it will be created with the `0777 &~ umask` permission.
```
Comment (joaofnfernandes) [inline registry/storage-drivers/filesystem.md]: Noob question. Is `0777 &~ umask` an actual command or are you just trying to signal that the actual permission used is the result of doing a logic AND of `0777` and whatever `umask` is set with?

Comment (stevvooe) [reply registry/storage-drivers/filesystem.md]: It might be better to link to a doc on umask and say that this honors `umask`.

Comment (AkihiroSuda) [reply registry/storage-drivers/filesystem.md]: updated, ptal

**Comment by joaofnfernandes** [comment]
Thanks for addressing the feedback. Let's ship it!

**Pull request merged**